ts_src/payments/p2tr_ns.ts

```diff
@@ -0,0 +1,142 @@
+import { bitcoin as BITCOIN_NETWORK } from '../networks';
+import { isXOnlyPoint } from '../schnorrBip340';
+import * as bscript from '../script';
+import { Payment, PaymentOpts, Stack } from './index';
+import * as lazy from './lazy';
+const OPS = bscript.OPS;
+const typef = require('typeforce');
+
+function stacksEqual(a: Buffer[], b: Buffer[]): boolean {
+  if (a.length !== b.length) return false;
+
+  return a.every((x, i) => {
+    return x.equals(b[i]);
+  });
+}
+
+// input: [signatures ...]
+// output: [pubKeys[0:n-1] OP_CHECKSIGVERIFY] pubKeys[n-1] OP_CHECKSIG
+export function p2tr_ns(a: Payment, opts?: PaymentOpts): Payment {
+  if (
+    !a.input &&
+    !a.output &&
+    !(a.pubkeys && a.pubkeys.length) &&
+    !a.signatures
+  )
+    throw new TypeError('Not enough data');
+  opts = Object.assign({ validate: true }, opts || {});
+
+  function isAcceptableSignature(x: Buffer | number): boolean {
+    return (
+      bscript.isCanonicalSchnorrSignature(x as Buffer) ||
+      (opts!.allowIncomplete && (x as number) === OPS.OP_0) !== undefined
+    );
```

Comment:
optional nit to avoid casts:
```suggestion
    if (Buffer.isBuffer(x)) {
      return bscript.isCanonicalSchnorrSignature(x);
    }
    return !!opts?.allowIncomplete ? x === OPS.OP_0 : false;
    );
```

I think there might also be some subtle logic issues here, eg (`allowIncomplete: false` and `x === OPS.OP_0`) would return true since `false !== undefined`.

Reply:
Using a slightly tweaked version of this, thanks!